File: Introduction to Programming/codeForces/J_Lucky_Division.py
val = int(input())

lc_num = [4,7]

i=8
while i!=1000:
    
    val1 = str(i)
    if all(i in ['4','7'] for i in val1) :
        lc_num.append(int(i))
   
    i+=1


flag =0
for i in lc_num:
    
    if val%i==0:
        flag=1
        break
  
        
if flag==1:
    print("YES")
else:
    print("NO")

File: Introduction to Programming/codeForces/J_Lucky_Division.c
#include <stdio.h>
#include <math.h>
#include <string.h>
int main(){

    char arr[1005];
    scanf("%s",arr);
    int i=0;
    int flag = 1;
    while (arr[i]!='\0')
    {
        int val = arr[i]-'0';
        
        if(val!=4){
            

            if(val!=7){
                flag=0;
                break;
            }
            
        }
        i++;
    }

    if(flag==1)printf("YES");
    else{

        int i=0;
        int len = strlen(arr)-1;

        int val =0;
        while (arr[i]!='\0'){
            int v = arr[i]-'0';

            int mul = pow(10,len);
            int final_val = mul*v;
           
           val+=final_val;
           
            len--;
            i++;
        }
        if(val%4==0 && val%7==0)printf("YES");
        else if(val%4==0)printf("YES");
        else if(val%7==0)printf("YES");
        else printf("NO");
    }

        

    
    return 0;
}